src/QtAVPlayer/qavplayer.cpp

```diff
@@ -316,6 +316,16 @@ void QAVPlayerPrivate::terminate()
     setState(QAVPlayer::StoppedState);
     quit = true;
     wait(false);
+
+    if (dev)
+        dev->abort(true);
+    demuxer.abort();
+    demuxerFuture.waitForFinished();
+    loaderFuture.waitForFinished();
+    videoPlayFuture.waitForFinished();
+    audioPlayFuture.waitForFinished();
+    demuxer.abort(false);
+
     videoFrameRate = 0.0;
     videoQueue.clear();
```

Comment:
as I remember, first we clear queues and after we wait that threads finished to consume all frames.